(PDB-2777) basic-persistence: with and w/o meta

rbrw opened the pull request:
The uncontended-performance test already tests the "no metadata" case to some extent too, so perhaps with this we're covered well enough for now...

rbrw committed:
(PDB-2777) Fix README underscores

rbrw committed:
(PDB-2777) queue-test: call 2 arg store if no meta

Don't call (store q s m) when there's no metadata, call the (store q s)
variant.

rbrw committed:
(PDB-2777) queue-test: check full metadata string

senior commented on test/puppetlabs/stockpile/queue_test.clj:
```diff
@@ -101,18 +102,32 @@
-             (is (= #{entry-2 entry-1} read-entries))
-             (is (= "foo" (slurp-entry q entry-1)))
-             (is (= "bar" (slurp-entry q entry-2))))))))))
+           ent-1-id (let [q (stock/create qdir)
```
It seems like the complexity of this test is rooted in trying to test two separate things in one test. If I'm understanding this correctly, ent-1-id is testing the storage of a stream without a metadata-string and ent2-id includes a metadata string. Seems like it would make the tests simpler to have them be separate (i.e. two separate testing clauses or two different deftests). Is there a reason to keep them together?

rbrw replied:
Possibly, though this is also testing reduce with more than one item (with and without metadata) along the way; I can certainly disentangle things and see if the (expanded) result's preferable.

senior reviewed (changes requested):
Looks good to me, just a minor comment on the interleaving of metadata and no-metadata tests

rbrw committed:
(PDB-2777) basic-persistence: with and w/o meta

Test one entry with metadata and one without, adjust store-str to
validate the resulting entry, and test reduce with an empty queue.
